File: src/algorithms/sort/ts/selectionSort.ts
export function selectionSort(values: number[]): void {
    if (values.length <= 1) {
        return
    }

    for (let i = 0; i < values.length - 1; i++) {
        let selectedIndex = i

        for (let j = i + 1; j < values.length; j++) {
            if (values[j] < values[selectedIndex]) {
                selectedIndex = j
            }
        }

        const temp = values[i]
        values[i] = values[selectedIndex]
        values[selectedIndex] = temp
    }
}

File: src/algorithms/sort/c/selectionSort.c
#include <stdio.h>

int selectionSort(int *arr, int arrLength){
    if (arrLength <= 0 ){
        printf("Array length must be greater than 0");
        return -1;
    } else if (arrLength == 1){
        return 0;
    }

    for (int i=0; i<arrLength - 1; i++){
        int selectedIndex = i;

        for (int j=i+1; j<arrLength; j++){
            if (arr[j] < arr[selectedIndex]){
                selectedIndex = j;
            }
        }

        int temp = arr[i];
        arr[i] = arr[selectedIndex];
        arr[selectedIndex] = temp;
    }

    return 0;
}
